src/main/java/com/salesforce/storm/spout/dynamic/VirtualSpout.java

```diff
@@ -592,13 +564,15 @@ private void attemptToComplete() {
         // Compare it against our ending state
         for (final ConsumerPartition consumerPartition: currentState.getConsumerPartitions()) {
             // currentOffset contains the last "committed" offset our consumer has fully processed
-            final long currentOffset = currentState.getOffsetForNamespaceAndPartition(consumerPartition);
```

Comment:
Curious as to what changed here in this logic that requires this swap?  I think these conditions need test cases.